dist/stat_data.py

```diff
@@ -498,6 +508,28 @@ def __init__(self, name, desc, flags=''):
     CacheStat('cache_write_restore', 'pages written requiring in-memory restoration'),
 
     ##########################################
+    # Cache content statistics
+    ##########################################
+    CacheWalkStat('cache_state_avg_written_size', 'Average on-disk page image  size seen', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_count_memory', 'Number of pages that have never been written from cache', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_count_not_queueable', 'Number of pages that could not be queued for eviction', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_count_queued', 'Number of pages currently queued for eviction', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_count_smaller_alloc_size', 'Number of on-disk page image sizes smaller than a single allocation unit', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_gen_avg_gap', 'Average difference between current eviction generation when the page was last considered', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_gen_current', 'Current eviction generation', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_gen_max_gap', 'Maximum difference between current eviction generation when the page was last considered', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_max_pagesize', 'Maximum page size seen', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_min_written_size', 'Minimum on-disk page image size seen', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_pages', 'Total number of pages currently in cache', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_pages_clean', 'Number of clean pages currently in cache', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_pages_dirty', 'Number of dirty pages currently in cache', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_pages_internal', 'Number of internal pages currently in cache', 'no_clear,no_scale'),
+    CacheWalkStat('cache_state_pages_leaf', 'Number of leaf pages currently in cache', 'no_clear,no_scale'),
```

Comment:
We don't currently, but it would be simple enough to add something into the cache tracking functions in ``btree.i`` - would you like me to do that instead?